schemas/package.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: package
+  title: Package
+  group: 2
+  short: These fields contain information about an installed software package.
+  description: >
+    These fields contain information about an installed software package. It contains general information about a
+    package, such as name, version or size. It also contains installation details, such as time, URL or
+    user who installed the package.
+  type: group
+  fields:
+
+    - name: artifact_id
```

Comment:
Agree to use `package.id`. 

It is meant to be usable for [entity_id](https://www.elastic.co/guide/en/beats/auditbeat/current/exported-fields-system.html#_package). IMO it should be uniquer per package + host it is installed. So reinstalling the same package on the same host would result in the same ID. Which information exactly is used for that depends on the collector. 
I suggest to at least use `package.name` + `package.versions` + `host.name`. We should probably align in how to collect it between beats and APM agents (or server if it is computed by the server).

Reply:
On `entity_id`: I introduced it when creating dashboards for the Auditbeat system module (https://github.com/elastic/beats/issues/10463). Without it, there was no way to display a total count of installed packages across all hosts.

To be honest, today I'm not sure it was a good idea. It makes it hard to combine data from Beats (which could theoretically all have it, while in practice so far only the Auditbeat system module has it), with data from third parties that will almost certainly never have it.

And I'm not sure it's needed - the total number of packages across hosts might not be terribly important to most people. Events such as package installs, uninstalls, and upgrades are, but those don't need the entity Id.

So in short: I'm not sure we should be specifying this in ECS.

Reply:
My thoughts were that this might be interesting information to query for unique package installations in an easy way, e.g. to update an existing document rather than creating a new one, or skipping to store a doc if it is already there. This idea is not mature enough yet from my side. With your comment to rather remove it from ECS based on your experience with auditbeat, I suggest to remove for now, but revisit if it makes sense for APM use case before the package field definition goes GA.